src/rdkafka_topic.c

```diff
@@ -636,14 +640,19 @@ static int rd_kafka_toppar_leader_update (rd_kafka_topic_t *rkt,
                         leader_id, rktp->rktp_broker_id);
                 r = 0;
         } else {
-                rktp->rktp_leader_id = leader_id;
-                if (rktp->rktp_leader)
-                        rd_kafka_broker_destroy(rktp->rktp_leader);
-                if (leader)
-                        rd_kafka_broker_keep(leader);
-                rktp->rktp_leader = leader;
-                r = rd_kafka_toppar_broker_update(rktp, leader_id, leader,
-                                                  "leader updated");
+                if (rktp->rktp_leader_id != leader_id) {
```

Comment:
ok, updated. I hate fiddling with this logic though - and you've previously told me off for making changes unrelated to incrreb in this PR, since we want to be able to point to specific PRs in the event of a screw up.